pennylane/qaoa/cost.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2018-2020 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+This file contains functions that generate QAOA cost Hamiltonians corresponding to
+different optimization problems.
+"""
+
+import networkx as nx
+import pennylane as qml
+
+
+def maxcut(graph):
+    r"""Returns the QAOA cost Hamiltonian corresponding to the MaxCut problem, for a given graph.
+
+    The goal of the MaxCut problem for a particular graph is to find a partition of nodes into two sets,
+    such that the number of edges in the graph with endpoints in different sets is maximized. Formally,
+    we wish to find the `cut of the graph <https://en.wikipedia.org/wiki/Cut_(graph_theory)>`__ such
+    that the number of edges crossing the cut is maximized.
+
+    The MaxCut Hamiltonian is defined as:
+
+    .. math:: H_C \ = \ \frac{1}{2} \displaystyle\sum_{(i, j) \in E(G)} \big( Z_i Z_j \ - \ \mathbb{I} \big)
+
+    where :math:`G` is some graph, :math:`\mathbb{I}` is the identity, and :math:`Z_i` and :math:`Z_j` are
+    the Pauli-Z operators on the :math:`i`-th and :math:`j`-th wire respectively.
+
+    .. note::
+
+        **Recommended mixer Hamiltonian:**
+            :func:`~.qaoa.x_mixer`
+
+        **Recommended initialization circuit:**
+            Even superposition over all basis states
+
+    Args:
+        graph (nx.Graph): a graph defining the pairs of wires on which each term of the Hamiltonian acts
+
+    Returns:
+        .Hamiltonian:
+
+    **Example**
+
+    >>> graph = nx.Graph([(0, 1), (1, 2)])
+    >>> cost_h = qml.qaoa.maxcut(graph)
+    (-0.5) [I0 I1] + (0.5) [Z0 Z1] + (-0.5) [I1 I2] + (0.5) [Z1 Z2]
+    """
```

Comment:
![image](https://user-images.githubusercontent.com/49409390/89054089-3b941580-d326-11ea-9610-abd2ad4c0f56.png)
The print is still needed